src/containers/ClusterTriggerBinding/ClusterTriggerBinding.js

```diff
@@ -11,123 +11,107 @@ See the License for the specific language governing permissions and
-    const {
-      match: prevMatch,
-      webSocketConnected: prevWebSocketConnected
-    } = prevProps;
-    const {
-      clusterTriggerBindingName: prevClusterTriggerBindingName
-    } = prevMatch.params;
-
-    if (
-      clusterTriggerBindingName !== prevClusterTriggerBindingName ||
-      (webSocketConnected && prevWebSocketConnected === false)
-    ) {
-      this.fetchData();
-    }
-  }
+  }, []);
 
-  fetchData() {
-    const { match } = this.props;
-    const { clusterTriggerBindingName } = match.params;
-    this.props.fetchClusterTriggerBinding({
+  function fetchData() {
+    fetchClusterTriggerBinding({
       name: clusterTriggerBindingName
     });
   }
 
-  render() {
-    const {
-      clusterTriggerBinding,
-      intl,
-      error,
-      loading,
-      selectedNamespace,
-      view
-    } = this.props;
-
-    const headersForParameters = [
-      {
-        key: 'name',
-        header: intl.formatMessage({
-          id: 'dashboard.tableHeader.name',
-          defaultMessage: 'Name'
-        })
-      },
-      {
-        key: 'value',
-        header: intl.formatMessage({
-          id: 'dashboard.tableHeader.value',
-          defaultMessage: 'Value'
-        })
-      }
-    ];
-
-    const rowsForParameters =
-      clusterTriggerBinding?.spec.params.map(({ name, value }) => ({
-        id: name,
-        name,
-        value
-      })) || [];
-
-    const emptyTextMessage = intl.formatMessage({
-      id: 'dashboard.clusterTriggerBinding.noParams',
-      defaultMessage: 'No parameters found for this ClusterTriggerBinding.'
-    });
+  useEffect(() => {
+    fetchData();
+  }, [clusterTriggerBindingName]);
+
+  useWebSocketReconnected(fetchData, webSocketConnected);
+
+  const headersForParameters = [
+    {
+      key: 'name',
+      header: intl.formatMessage({
+        id: 'dashboard.tableHeader.name',
+        defaultMessage: 'Name'
+      })
+    },
+    {
+      key: 'value',
+      header: intl.formatMessage({
+        id: 'dashboard.tableHeader.value',
+        defaultMessage: 'Value'
+      })
+    }
+  ];
+
+  const rowsForParameters =
+    clusterTriggerBinding?.spec.params.map(({ name, value }) => ({
+      id: name,
+      name,
+      value
+    })) || [];
+
+  const emptyTextMessage = intl.formatMessage({
+    id: 'dashboard.clusterTriggerBinding.noParams',
+    defaultMessage: 'No parameters found for this ClusterTriggerBinding.'
+  });
 
-    return (
-      <ResourceDetails
-        error={error}
-        loading={loading}
-        onViewChange={getViewChangeHandler(this.props)}
-        resource={clusterTriggerBinding}
-        view={view}
-      >
-        <Table
-          title={intl.formatMessage({
-            id: 'dashboard.parameters.title',
-            defaultMessage: 'Parameters'
-          })}
-          headers={headersForParameters}
-          rows={rowsForParameters}
-          size="short"
-          selectedNamespace={selectedNamespace}
```

Comment:
Question, was passing this prop actually a bug?

Reply:
The `selectedNamespace` prop was never actually set (because it's a cluster-scoped resource) so this was always undefined. Removed it here to avoid confusion. In fact I'll probably update the `Table` component in a future PR to accept a single `emptyText` prop since it shouldn't really be aware of namespaces at all.